ui/src/app/records/custom-editor/circulation-settings/circulation-policy.service.ts

```diff
@@ -22,56 +22,38 @@ import { Injectable } from '@angular/core';
 import { forkJoin, of } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { CirculationPolicy } from './circulation-policy';
-import { ItemTypeService, PatronTypeService, ApiService, cleanDictKeys, _ } from '@app/core';
+import { ApiService, cleanDictKeys, _ } from '@app/core';
 import { RecordsService } from '@app/records/records.service';
 import { Router } from '@angular/router';
 import { ToastrService } from 'ngx-toastr';
 
-const httpOptions = {
-  headers: new HttpHeaders({
-    'Accept': 'application/json',
-    'Content-Type': 'application/json'
-  })
-};
-
 @Injectable()
 export class CirculationPolicyService {
 
   constructor(
     private router: Router,
     private apiService: ApiService,
     private client: HttpClient,
-    private itemTypeService: ItemTypeService,
-    private patronTypeService: PatronTypeService,
     private recordsService: RecordsService,
     private toastService: ToastrService
   ) { }
 
   loadOrCreateCirculationPolicy(pid: number = null) {
     if (pid) {
-      return this.client.get<any>(
-        this.apiService.getApiEntryPointByType('circ_policies') + pid,
-        httpOptions
-      ).pipe(
+      return this.recordsService.getRecord('circ_policies', '' + pid)
+      .pipe(
         map(data => new CirculationPolicy(data.metadata))
       );
     } else {
       return of(new CirculationPolicy());
     }
   }
 
-  loadAllCirculationPolicy() {
-    return this.client.get(
-      this.apiService.getApiEntryPointByType('circ_policies'),
-      httpOptions
-    );
-  }
-
   loadAllItemTypesPatronTypesCirculationPolicies() {
     return forkJoin(
-      this.itemTypeService.itemTypes(),
-      this.patronTypeService.patronTypes(),
-      this.loadAllCirculationPolicy()
+      this.recordsService.getRecords('global', 'item_types', 1, 9999),
```

Comment:
these 9999 can be replaced by a variable or better in a configuration service (can be recordsService for the moment).

Reply:
Added 2 constants into `RecordServices` class : DEFAULT_REST_RESULTS_SIZE, MAX_REST_RESULTS_SIZE.